File: lsp/intellij-plugin/src/main/java/community/flock/wire_spec/lsp/intellij_plugin/FileType.java
package community.flock.wire_spec.lsp.intellij_plugin;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class FileType extends LanguageFileType {

    public static final FileType INSTANCE = new FileType();

    private FileType() {
        super(Language.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "wire-spec";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "wire-spec language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "ws";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return Icons.FILE;
    }

}

File: src/ide/intellij-plugin/src/main/kotlin/community/flock/wirespec/lsp/intellij_plugin/FileType.kt
package community.flock.wirespec.lsp.intellij_plugin

import com.intellij.openapi.fileTypes.LanguageFileType

object FileType : LanguageFileType(Language) {
    override fun getName() = "wirespec"

    override fun getDescription() = "Wirespec language file"

    override fun getDefaultExtension() = "ws"

    override fun getIcon() = Icons.FILE
}
